lib/compiler/src/compiler.rs

```diff
@@ -16,6 +17,17 @@ use wasmer_types::entity::PrimaryMap;
 use wasmer_types::{Features, FunctionIndex, LocalFunctionIndex, SignatureIndex};
 use wasmparser::{Validator, WasmFeatures};
 
+/// The output of the experimental native compilation.
+/// It returns the object files that the linker will link together after
+/// along with the frame infos necessary for fully traps support.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ExperimentalNativeCompilation {
+    /// The kind of emitted object file
+    pub object_files: Vec<Vec<u8>>,
+    /// The frame infos of the files
+    pub frame_infos: PrimaryMap<LocalFunctionIndex, CompiledFunctionFrameInfo>,
+}
```

Comment:
How is the frame info not part of the object file??

Reply:
This frame info’s are Wasmer ones, where we keep track of everything.
They live in the CompiledModule info which live in the metadata (which is a proprietary format of Wasmer). That’s why it has to live here and not the object file